src/test/java/com/exasol/adapter/dialects/ScalarFunctionsParameterCache.java

```diff
@@ -0,0 +1,62 @@
+package com.exasol.adapter.dialects;
+
+import java.io.*;
+import java.util.*;
+
+import org.yaml.snakeyaml.Yaml;
+
+import com.exasol.errorreporting.ExaError;
+
+/**
+ * This class caches the parameters for runs that were passed the tests. By that in upcoming test runs can only test
+ * these runs and by that are a lot faster.
+ */
+public class ScalarFunctionsParameterCache {
+    private static final String CACHE_FILE_NAME = "src/test/resources/scalarFunctionsParameterCache.yml";
+    private final Map<String, List<String>> parameterCache;
+    private final Yaml yaml;
+
+    public ScalarFunctionsParameterCache() {
+        try {
+            this.yaml = new Yaml();
+            this.parameterCache = loadParameterCache();
+        } catch (final FileNotFoundException exception) {
+            throw new IllegalStateException(ExaError.messageBuilder("E-PGVS-10")
```

Comment:
Do we use Error builder in the tests too? Then I saw a few error messages without Error builder up there, we should unify them

Reply:
I asked @redcatbear if I should and he said yes...

Reply:
Unifying it is a good idea. I changed all exception now to use the builder